File: src/main/java/com/saiph/application/GestionRH/Domain/dto/DepartementDto.java
package com.saiph.application.GestionRH.Domain.dto;


import com.saiph.application.GestionRH.Domain.entities.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DepartementDto extends GenericDto {


    @NonNull
    @Column(nullable = false, unique = true)
    private String name;

    private UserDto manager;

    @OneToMany(fetch = FetchType.EAGER)
    private List<UserDto> ListEmploye = new ArrayList<>();

}

File: Gestion-Rh-UI/src/app/services/models/departement-dto.ts
/* tslint:disable */
/* eslint-disable */
import { UserDto } from '../models/user-dto';
export interface DepartementDto {
  createdBy?: string;
  createdDate?: string;
  id?: number;
  lastModifiedBy?: string;
  lastModifiedDate?: string;
  listEmploye?: Array<UserDto>;
  manager?: UserDto;
  name?: string;
}
